packages/@aws-cdk/aws-codepipeline-actions/lib/stepfunctions/invoke-action.ts

```diff
@@ -0,0 +1,149 @@
+}
+
+/**
+ * Construction properties of the {@link StepFunctionsInvokeAction StepFunction Invoke Action}.
+ */
+export interface StepFunctionsInvokeActionProps extends codepipeline.CommonAwsActionProps {
+  /**
+   * The optional input Artifact of the Action.
+   * If InputType is set to FilePath, this artifact is required 
+   * and is used to source the input for the state machine execution.
+   * 
+   * @default the Action will not have any inputs
+   * @see https://docs.aws.amazon.com/codepipeline/latest/userguide/action-reference-StepFunctions.html#action-reference-StepFunctions-example
+   */
+  readonly input?: codepipeline.Artifact;
+ 
+  /**
+   * The optional output Artifact of the Action.
+   *
+   * @default the Action will not have any outputs
+   */
+  readonly output?: codepipeline.Artifact;
+ 
+  /**
+   * The state machine to invoke.
+   */
+  readonly stateMachine: stepfunction.IStateMachine;
+ 
+  /**
+   * Optional StateMachine InputType
+   * InputType can be Literal or FilePath
+   *
+   * @default - Literal
+   */
+  readonly stateMachineInputType?: StateMachineInputType;
+ 
+  /**y
+   * When InputType is set to Literal (default), this field is optional.
+   * If provided, the Input field is used directly as the input for the state machine execution.
+   * Otherwise, the state machine is invoked with an empty JSON object {}.
+   * 
+   * When InputType is set to FilePath, this field is required.
+   * An input artifact is also required when InputType is set to FilePath.
+   * 
+   * @default - none
+   */
+  readonly stateMachineInput?: string | object;
+ 
+ 
+  /**
+   * Prefix (optinoal)
+   *
+   * By default, the action execution ID is used as the state machine execution name. 
+   * If a prefix is provided, it is prepended to the action execution ID with a hyphen and 
+   * together used as the state machine execution name.
+   */
+  readonly executionPrefixName?: string;
+}
+ 
+/**
+ * CodePipeline invoke Action that is provided by an AWS Step Function.
+ */
+export class StepFunctionsInvokeAction extends Action {
+  private readonly props: StepFunctionsInvokeActionProps;
+ 
+  constructor(props: StepFunctionsInvokeActionProps) {
+    super({
+      ...props,
+      resource: props.stateMachine,
+      category: codepipeline.ActionCategory.INVOKE,
+      provider: 'StepFunctions',
+      artifactBounds: {
+        minInputs: 0,
+        maxInputs: 1,
+        minOutputs: 0,
+        maxOutputs: 1,
+      },
+    });
+
+    //StateMachineInput is required when the StateMachineInputType is set as FilePath
+    if ( props.stateMachineInputType == StateMachineInputType.FILEPATH && (!props.stateMachineInput)) {
+      throw new Error('Input type FilePath was specified, but no filepath was provided');
+    }
+
+    //Input Artifact is required when the StateMachineInputType is set as FilePath
+    if ( props.stateMachineInputType == StateMachineInputType.FILEPATH && (!this.actionProperties.inputs)) {
```

Comment:
I think this would be easier to read like this:

```
if (props.stateMachineInputType == StateMachineInputType.FILEPATH) {
   if (!props.stateMachineInput) {

   }
   if (!this.actionProperties.inputs) {

   }
}
```